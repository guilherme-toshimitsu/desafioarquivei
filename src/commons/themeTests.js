import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
