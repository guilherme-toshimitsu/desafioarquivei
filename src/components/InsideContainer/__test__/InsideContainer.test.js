import React from "react";
import InsideContainer from "../";
import { renderWithTheme } from "@commons/themeTests";

describe("<InsideContainer />", () => {
  it("creates a InsideContainer component", () => {
    const wrapper = renderWithTheme(<InsideContainer />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
