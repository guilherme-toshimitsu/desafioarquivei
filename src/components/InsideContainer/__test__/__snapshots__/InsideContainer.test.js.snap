// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InsideContainer /> creates a InsideContainer component 1`] = `
<DocumentFragment>
  <div
    class="InsideContainer__Container-sc-1t900ay-0 cAJihm"
  />
</DocumentFragment>
`;
